qwandaq/src/main/java/life/genny/qwandaq/utils/QwandaUtils.java

```diff
@@ -544,36 +544,66 @@ public Ask generateAskGroupUsingBaseEntity(BaseEntity baseEntity) {
+//				.filter(ea -> defBE.containsEntityAttribute("ATT_" + ea.getAttributeCode()))
 				.forEach((ea) -> {
+					String attributeCode = StringUtils.removeStart(ea.getAttributeCode(),"ATT_");
+					attributeCode = StringUtils.removeStart(attributeCode,"DEP_");
+					attributeCode = StringUtils.removeStart(attributeCode,"DFT_");
+					attributeCode = StringUtils.removeStart(attributeCode,"UNQ_");
+
+					Optional<EntityAttribute> baseEA = baseEntity.findEntityAttribute(attributeCode);
+
+					if (!baseEA.isPresent()) {
+						String questionCode = attributeCode;
+						questionCode = "QUE_" + StringUtils.removeStart(StringUtils.removeStart(questionCode, "PRI_"), "LNK_");
+						Question childQues = null;
+
+						try {
+							Attribute attribute = databaseUtils.findAttributeByCode(userToken.getRealm(), attributeCode);
+							childQues = new Question(questionCode, attribute.getName(), attribute);
+						} catch (Exception ex) {
+							childQues = new Question(questionCode, ea.getAttribute().getName(), ea.getAttribute());
+						}
 
-					String questionCode = "QUE_"
-							+ StringUtils.removeStart(StringUtils.removeStart(ea.getAttributeCode(), "PRI_"), "LNK_");
+						Ask childAsk = new Ask(childQues, userToken.getUserCode(), attributeCode);
 
-					Question childQues = new Question(questionCode, ea.getAttribute().getName(), ea.getAttribute());
-					Ask childAsk = new Ask(childQues, userToken.getUserCode(), baseEntity.getCode());
+						childAsks.add(childAsk);
+					} else {
+						EntityAttribute baseAttrVal = baseEA.get();
+						String questionCode = "QUE_"
+								+ StringUtils.removeStart(StringUtils.removeStart(baseAttrVal.getAttributeCode(), "PRI_"), "LNK_");
 
-					childAsks.add(childAsk);
+						Question childQues = new Question(questionCode, baseAttrVal.getAttribute().getName(), baseEA.get().getAttribute());
+						Ask childAsk = new Ask(childQues, userToken.getUserCode(), baseEntity.getCode());
 
-					if (ea.getAttributeCode().startsWith("LNK_")) {
-						if (ea.getValueString() != null) {
+						childAsks.add(childAsk);
 
-							String[] codes = BaseEntityUtils.cleanUpAttributeValue(ea.getValueString()).split(",");
+						if (baseAttrVal.getAttributeCode().startsWith("LNK_")) {
+							if (baseAttrVal.getValueString() != null) {
 
-							for (String code : codes) {
-								try {
-									BaseEntity link = beUtils.getBaseEntityByCode(code);
+								if (baseAttrVal.getValueString().isEmpty()) {
+									BaseEntity link = beUtils.getBaseEntityByCode(baseAttrVal.getAttributeCode());
 									entityMessage.add(link);
-								} catch (Exception ex) {
-									log.error(ex);
+								} else {
+									String[] codes = BaseEntityUtils.cleanUpAttributeValue(baseAttrVal.getValueString()).split(",");
+
+									for (String code : codes) {
+										try {
+											BaseEntity link = beUtils.getBaseEntityByCode(code);
+											entityMessage.add(link);
+										} catch (Exception ex) {
```

Comment:
Use `beUtils.getBaseEntity()`, and don't catch the exception as we would rather a fail fast approach.

Reply:
removed try catch